### ISSUE TITLE
Add Support for Custom Header Panel (Custom Widget) in showBoardDateTimePicker

### ISSUE THREAD

**Issue by flashback19** [mytooyo/board_datetime_picker#46]
It would be great to have the ability to add a custom header panel (custom widget) in the showBoardDateTimePicker dialog. This feature would allow developers to provide their own header design, adding more flexibility and enabling unique use cases such as adding additional information, icons or custom actions at the top of the picker.

**Proposed API Changes:**

- Introduce a new optional parameter headerWidget to showBoardDateTimePicker.
- This parameter will accept a Widget that will be displayed as the header of the dialog.
- The default header could still be used if the parameter is not provided, ensuring backward compatibility.

**Benefits:**

- Allows for more customization options and a richer user experience.
- Enables developers to provide more context or control directly in the header of the date-time picker.
- Gives the flexibility to modify the look and feel to match the design system of the app.

**Example Usage:**

```
await showBoardDateTimePicker(
      context: context,
      pickerType: DateTimePickerType.date,      
      headerWidget: MyCustomHeader(),
    );
```

**Comment by mytooyo** [comment]
@flashback19 
Good feature about the custom header.
The current header shows a calendar switch button and a close button,
so need to consider what to do about it.

**Comment by mytooyo** [comment]
@flashback19 

Is the header correct in the white card widget where the calendar button is currently placed?
 If that is the case, then using a custom header would make it impossible to place a calendar button.
 I can solve this by making the button a class and making it available within the custom header, but it will take time..

**Comment by mytooyo** [comment]
@flashback19 
The ability to add custom headers has been released in v2.1.3.
Along with the custom header feature, I have also added a parameter that allows you to set the top margin of the menu bar. Use it to adjust the header display.

```dart
await showBoardDateTimePicker(
  context: context,
  pickerType: pickerType,
  headerWidget: Container(height: 40, color: Colors.white), // <- added parameter
  options: BoardDateTimeOptions(
    topMargin: 8,  // <- added parameter (default is 20)
  ),
);
```
